File: src/main/java/io/github/faecraft/heartshapedbox/math/two_d/Line.java
package io.github.faecraft.heartshapedbox.math.two_d;

import net.minecraft.util.math.Vec2f;

public class Line {
    final Vec2f start;
    final double slope;
    
    final float b;
    
    public Line(Vec2f start, double slope) {
        this.start = start;
        this.slope = slope;
        this.b = (float)(start.y - (slope * start.x));
    }
    
    public double solveForX(double y) {
        return (y - b) / slope;
    }
    
    public double solveForY(double x) {
        return (slope * x) + b;
    }
}

File: src/main/kotlin/io/github/faecraft/heartshapedbox/math/two_d/Line.kt
package io.github.faecraft.heartshapedbox.math.two_d

import net.minecraft.util.math.Vec2f

class Line(val start: Vec2f, val slope: Double) {
    val b: Float
    fun solveForX(y: Double): Double {
        return (y - b) / slope
    }

    fun solveForY(x: Double): Double {
        return slope * x + b
    }

    init {
        b = (start.y - slope * start.x).toFloat()
    }
}
